src/Analyzers/CSharp/CodeFixes/UseInterpolatedString/CSharpUseInterpolatedStringCodeFixProvider.cs

```diff
@@ -0,0 +1,103 @@
+        var node = root.FindNode(span, getInnermostNodeForTie: true) as LiteralExpressionSyntax;
+        if (node?.Span != span || !node.IsKind(SyntaxKind.StringLiteralExpression))
+            return;
+
+        var newNode = CreateInterpolatedString(node);
+        editor.ReplaceNode(node, newNode);
+    }
+
+    private static InterpolatedStringExpressionSyntax CreateInterpolatedString(SyntaxNode stringLiteralExpressionNode)
+    {
+        var token = stringLiteralExpressionNode.GetFirstToken();
+        var originalTokenKind = token.Kind();
+        var isVerbatimString = token.IsVerbatimStringLiteral();
+        var newTokenKind = GetKind(originalTokenKind);
+        var startTokenKind = GetStartToken(originalTokenKind, isVerbatimString);
+        var text = GetTextWithoutQuotes(token, isVerbatimString);
+        var endTokenKind = GetEndToken(originalTokenKind);
+
+        var newNode = SyntaxFactory.InterpolatedStringText(
+            SyntaxFactory.Token(
+                leading: SyntaxTriviaList.Empty,
+                kind: newTokenKind,
+                text: text,
+                valueText: text,
+                trailing: SyntaxTriviaList.Empty))
+            .WithTriviaFrom(stringLiteralExpressionNode);
+
+        return SyntaxFactory.InterpolatedStringExpression(
+            stringStartToken: SyntaxFactory.Token(startTokenKind),
+            contents: [newNode],
+            stringEndToken: SyntaxFactory.Token(endTokenKind));
+    }
+
+    private static SyntaxKind GetKind(SyntaxKind kind) => kind switch
+    {
+        SyntaxKind.SingleLineRawStringLiteralToken => SyntaxKind.InterpolatedStringTextToken,
+        SyntaxKind.MultiLineRawStringLiteralToken => SyntaxKind.InterpolatedStringTextToken,
+        _ => SyntaxKind.InterpolatedStringTextToken,
+    };
+
+    private static SyntaxKind GetStartToken(SyntaxKind kind, bool isVerbatimString) => kind switch
+    {
+        SyntaxKind.SingleLineRawStringLiteralToken => SyntaxKind.InterpolatedSingleLineRawStringStartToken,
+        SyntaxKind.MultiLineRawStringLiteralToken => SyntaxKind.InterpolatedMultiLineRawStringStartToken,
+        _ => isVerbatimString ? SyntaxKind.InterpolatedVerbatimStringStartToken : SyntaxKind.InterpolatedStringStartToken,
+    };
+
+    private static string GetTextWithoutQuotes(SyntaxToken token, bool isVerbatimString)
+    {
+        var text = token.Text.Replace("{", "{{").Replace("}", "}}");
+        var kind = token.Kind();
+        var startIndex = GetStartIndex(kind, isVerbatimString);
+        var endIndex = GetEndIndex(kind);
+        return text[startIndex..^endIndex];
+    }
+
+    private static SyntaxKind GetEndToken(SyntaxKind kind) => kind switch
+    {
+        SyntaxKind.SingleLineRawStringLiteralToken => SyntaxKind.InterpolatedRawStringEndToken,
+        SyntaxKind.MultiLineRawStringLiteralToken => SyntaxKind.InterpolatedRawStringEndToken,
```

Comment:
so the problem here is that's not sufficient to just return a kind here.  Basically, many tokens in teh language have a 1:1 mapping between their kind and the exact text that must appear in the token contents.  For example the ClassKeyword kind exactly maps to `class`.  This doesn't apply to `InterpolatedRawStringEndToken` because the end token contents are *variable*.  For example, a raw string can end with `"""` or `""""` or `"""""` or any number of quotes.  Similarly, an interplated raw start token could be `$"""` `$""""` `$"""""` and so on.  It can also be `$$"""` `$$$"""` and so on.  Basically it can start with some number of dollars, at least 3 quotes, and then must end with the same number of quotes that it started with.

Reply:
@CyrusNajmabadi
1. In that case, should I manually find out the start token (whether it be three double quotes or four double quotes or more) and have the same number of double quotes in the end token?
2. Once I determine that, would the next step be `SyntaxFactory.ParseToken("\"\"\"\"")`?

Reply:
1. Yes. You'll need to figure it out. 
2.  When creating a token, you can specify the text content. You don't need to parse it. 

Thanks!

Reply:
@CyrusNajmabadi  Taking a step back, I see in https://github.com/dotnet/roslyn/blob/main/src%2FFeatures%2FCore%2FPortable%2FConvertToInterpolatedString%2FConvertRegularStringToInterpolatedStringRefactoringProvider.cs#L44-L45 that raw string literals are not touched by the existing refactoring. Should I maintain this behavior?

Context: I was debugging raw string literal tests 
![image](https://github.com/dotnet/roslyn/assets/7676354/83eb4110-996b-44c0-86aa-891884586609)
and realized these tests didn't exist in https://github.com/dotnet/roslyn/blob/0d1f9fea04892d9e65e9a780bb773a4d5ed18f8e/src/Features/CSharpTest/ConvertToInterpolatedString/ConvertRegularStringToInterpolatedStringTests.cs.

Reply:
> that raw string literals are not touched by the existing refactoring. Should I maintain this behavior?

Up to you.  I didn't do it because there wasn't really the need for such an upgrade path.

Put another way, when the language started, we didn't have interpolated strings.  So the idea of these features was to take the older patterns people would use to make strings, and move them to use the nicer new stuff.

But raw-strings *are* hte nice new stuff.  And when we did raw strings, we made sure we had normal raw strings and interpolated raw strings at the same time.  So there's a lot less need to go "oh... someone built a complex expr with raw strings, that they now want to convert to an interpolated form".  Instead, generally, people just start with the interpolated form in the first place.

Reply:
Okay, I guess I misunderstood the intent of the refactoring then. I thought this feature would be useful when you wanted to convert a regular string to an interpolated string because you now want to use variables in it, as in [here](https://github.com/nunit/nunit.analyzers/pull/721/files#diff-4eb6ac27ac8fd046156ed20ecb78685c3f0f8a330f3a88e6084462494026d098R140-R145). Since that can be true for raw string literals, I was going to do the same for them.

If it's up to me, I think I'll do a little more of poking around to see if I can do this. It didn't seem straightforward on a cursory look.